lib/twilio/audiohelper.ts

```diff
@@ -275,6 +275,16 @@ class AudioHelper extends EventEmitter {
     if (isEnumerationSupported) {
       this._initializeEnumeration();
     }
+
+    navigator.permissions.query({ name: 'microphone' as PermissionName }).then((microphonePermissionStatus) => {
+      if (microphonePermissionStatus.state !== 'granted') {
+        const handleStateChange = () => {
+          this._updateAvailableDevices();
+          microphonePermissionStatus.removeEventListener('change', this._updateAvailableDevices);
```

Comment:
```suggestion
          microphonePermissionStatus.removeEventListener('change', handleStateChange);
```

Reply:
We also need to call removeEventListener in the `_destroy()` function. Or else, this will cause a memory leak if the permission never changes after destroying the audio helper. To do this, you might want to save a reference to the microphonePermissionStatus and the change handler: `this._microphonePermissionStatus` and `this._onMicrophonePermissionStatusChanged`.